CMake/AbseilHelpers.cmake

```diff
@@ -62,7 +62,107 @@ function(absl_library)
   endif()
 endfunction()
 
+#
+# CMake function to imitate Bazel's cc_library rule.
+#
+# Parameters:
+# NAME: name of target (see Note)
+# HDRS: List of public header files for the library
+# SRCS: List of source files for the library
+# DEPS: List of other libraries to be linked in to the binary targets
+# COPTS: List of private compile options
+# DEFINES: List of public defines
+# LINKOPTS: List of link options
+# PUBLIC: Add this so that this library will be exported under absl:: (see Note).
+# TESTONLY: When added, this target will only be built if user passes -DBUILD_TESTING=ON to CMake.
+#
+# Note:
+#
+# By default, absl_cc_library will always create a library named absl_internal_${NAME},
+# which means other targets can only depend this library as absl_internal_${NAME}, not ${NAME}.
+# This is to reduce namespace pollution.
+#
+# absl_cc_library(
+#   NAME
+#     awesome_lib
+#   HDRS
+#     "a.h"
+#   SRCS
+#     "a.cc"
+# )
+# absl_cc_library(
+#   NAME
+#     fantastic_lib
+#   SRCS
+#     "b.cc"
+#   DEPS
+#     absl_internal_awesome_lib # not "awesome_lib"!
+# )
+#
+# If PUBLIC is set, absl_cc_library will also create an alias absl::${NAME}
```

Comment:
why don't using "absl_" prefix for PUBLIC enable libraries ?

Reply:
It is already there. The following code will create `absl_strings` *and* `absl::strings` (alias). CMake user should depend on `absl::strings` only.

```
absl_cc_library(
  NAME
    strings
  PUBLIC
)
```

Reply:
no it will create `absl_internal_strings` *and* `absl::strings` line 125
you should have something like this:
```cmake
if (ABSL_CC_LIB_PUBLIC) 
  set(_NAME "absl_${ABSL_CC_LIB_NAME}")
else()
  set(_NAME "absl_internal_${ABSL_CC_LIB_NAME}")
endif()

```

Reply:
> no it will create absl_internal_strings and absl::strings line 125

Ah, I forgot that the prefix has changed to `absl_internal_` (it was initially just `absl_`, but Jonathan requested to always use `absl_internal_`).

Is it better for user to use `absl_strings` instead of `absl::strings`? Because I want to make it clear to CMake user that only `absl::` can be depended outside Abseil. (No strong objection here, just want your opinion).

Reply:
Since:
1) abseil force people to use abseil-cpp as `subdirectory()`,
So meta project who want to add an install rules will fail at configure time if they don't add `absl_string` to one `install(TARGETS ...` (that's why so much people want ).
Twist
2) target alias are forbidden in `install(TARGETS ....` command

-> `1 + 2` -> you can't integrate `abseil-cpp` in your project unless you never install it...

Reply:
Done.